### PR TITLE
feat: django setup

### PR DESCRIPTION
**Pull request by Thomas-Jean**
feat: user and basic crud example

feat: pytest up and working

feat: added linting and formating

fix: update depecated function

doc: update readme with setup instructions

ci: added basic ci pipeline

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,69 @@
...
-1. Select "theartofeducation" as the "Owner" of the repository
-1. Add the repository name
-1. Add a short description for the repository
-1. Select "Public" or "Private" as appropriate for the repository
-1. Click the "Create repository" button
-1. Update files and repository information as needed
-1. Once your repo is setup rebase `develop` on `main` and force push to the `origin`
-    * This ensures that `develop` is based on `master` so when you create a PR in
-     the future GitHub will allow it
+# AoEU API
```
Comment (theantichris) [inline README.md:1]: Needs to be AOEU for proper branding.

Great work on the setup instructions!

**Review by theantichris — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ryexley) [reply README.md:1]: 👍 for great setup instructions, totally agree ... really nicely done.

File: api/tenant_management/tests/models/user_test.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import re
+
+import pytest
+
+from ...models.user import User
+
+# @pytest.fixture
+# def alice(db) -> User:
+#     return
```
Comment (ryexley) [inline api/tenant_management/tests/models/user_test.py:9]: Is this some kind of setup directive block, or is this commented out code? Does it need to be here??

**Review by ryexley — APPROVED** [review]
Only the one question, other than that, looks like a really good start. Thanks for the documentation and the good example to draw from in this initial setup.

File: DECISION-LOG-BOOK.md [hunk 3]
```diff
@@ -33,7 +38,7 @@ Elixir
 Node
     + Team Familiarity
```
Comment (theantichris) [inline DECISION-LOG-BOOK.md:39]: These are still all be displayed as single lines in GitHub.

File: DECISION-LOG-BOOK.md [hunk 4]
```diff
@@ -10,15 +10,20 @@ Thomas, Christopher, Bob
 
 ### Context
 
-The current context and constraints during the decision.
+We are starting the construction of the new backend for AOEU and we want a distinct api layer
```
Comment (theantichris) [inline DECISION-LOG-BOOK.md:13]: All caps "API".

File: DECISION-LOG-BOOK.md [hunk 5]
```diff
@@ -10,15 +10,20 @@ Thomas, Christopher, Bob
...
-Python as the primary language for the API layer
+Python as the primary language for the API layer/gateway.
+We chose this as it would allow for a dynamic language at the edge to help manage external json.
+Additional focus on a stateless application pushing state to a database, cache, or other services.
 
 #### Why This Solution
 
-Tooling and support for REST patterns for building out the API
+Tooling and support for REST patterns for building out the API.
+Django Rest Framework has some patterns and libraries that we can use to build and validate requests.
```
Comment (theantichris) [inline DECISION-LOG-BOOK.md:25]: All caps "REST".

**Review by theantichris — CHANGES_REQUESTED** [review]

**Comment by theantichris** [comment]
I want to say I'm not being picky about API vs api but it wasn't consistent through the document.

File: api/tenant_management/tests/models/user_test.py (hunk 2 above)
Comment (Thomas-Jean) [reply api/tenant_management/tests/models/user_test.py]: This was a test fixture I forgot about. I've removed it.

**Review by theantichris — APPROVED** [review]

**Pull request merged**